extensions/parquet/table/src/main/java/io/deephaven/parquet/table/TypeInfos.java

```diff
@@ -93,9 +101,103 @@ private static TypeInfo lookupTypeInfo(
         return new ImmutablePair<>(SerializableCodec.class.getName(), null);
     }
 
+    static class PrecisionAndScale {
+        public final int precision;
+        public final int scale;
+
+        public PrecisionAndScale(final int precision, final int scale) {
+            this.precision = precision;
+            this.scale = scale;
+        }
+    }
+
+    private static PrecisionAndScale getPrecisionAndScale(final TrackingRowSet rowSet,
+            final ColumnSource<BigDecimal> source) {
+        final int sz = 4096;
+        final ChunkSource.GetContext context = source.makeGetContext(sz);
+        // we first compute max(precision - scale) and max(scale), which corresponds to max(digits left of the decimal
+        // point),
+        // max(digits right of the decimal point). Then we convert to (precision, scale) before returning.
+        int maxPrecisionMinusScale = 0;
+        int maxScale = 0;
+        try (final RowSequence.Iterator it = rowSet.getRowSequenceIterator()) {
+            final RowSequence ok = it.getNextRowSequenceWithLength(sz);
+            final ObjectChunk<BigDecimal, ? extends Values> chunk = source.getChunk(context, ok).asObjectChunk();
+            for (int i = 0; i < chunk.size(); ++i) {
```

Comment:
I wonder if it would be worth trying to assess whether you have a fixed width data set while doing this.
It's a little iffy. You can get `unscaledValue()` which is an immutable `BigInteger`, and check it's `bitLength()` and round up the nearest `byte`. But then you might need to adjust if you have more than one scale. And *sometimes* `unscaledValue()` allocates, other times it doesn't.

Reply:
I don't understand your suggestion.  I don't understand it at multiple levels actually.  I am not sure what we would be improving, nor how.  By saying "a fixed width data set" I am guessing you mean all the values have the same scale? (they may not have the same precision, eg, the precision of "1.23" is 3, and the precision of "12.34" is 4, etc).